packages/app-project/pages/project.js

```diff
@@ -1,15 +1,69 @@
+import { Grommet } from 'grommet'
+import { Provider } from 'mobx-react'
+import { getSnapshot } from 'mobx-state-tree'
+import PropTypes from 'prop-types'
+import React from 'react'
+
 import Head from '../components/head'
 import Nav from '../components/nav'
-import React from 'react'
-import { Grommet } from 'grommet'
+import ProjectHome from '../components/ProjectHome'
+import getProjectSlugFromURL from '../helpers/getProjectSlugFromURL'
+import initStore from '../stores'
+
+class ProjectPage extends React.Component {
+  // getInitialProps looks like a React lifecycle method, but is actually a
+  // Next.js method only called in page components. See https://git.io/vhZz9
+  static async getInitialProps ({ req }) {
+    const props = {
+      isServer: !!req
+    }
+
+    if (props.isServer) {
+      const store = initStore(props.isServer)
+      const slug = getProjectSlugFromURL(req.url)
+      await store.project.fetch(slug)
+      props.initialState = getSnapshot(store)
+    }
+    return props
+  }
+
+  constructor (props) {
+    super(props)
+    this.store = initStore(props.isServer, props.initialState)
+  }
+
+  componentDidUpdate (prevProps) {
+    const slug = getProjectSlugFromURL(this.props.url.asPath)
+    const prevSlug = getProjectSlugFromURL(prevProps.url.asPath)
+    if (slug !== prevSlug) {
+      this.store.project.fetch(slug)
```

Comment:
This isn't always firing correctly. I've navigated a few times between the projects and the home page and don't always see the project display name or see the wrong display name (the store still has the old project). There should probably be a spec for this? 

See wrong project display name:
<img width="567" alt="screen shot 2018-06-04 at 10 52 26 am" src="https://user-images.githubusercontent.com/5016731/40927874-77ff0c54-67e5-11e8-8e94-804bc76e287f.png">